src/main/java/org/folio/service/transactions/batch/BatchPaymentCreditService.java

```diff
@@ -90,8 +92,10 @@ private void updateEncumbranceToCancelTransaction(Transaction transaction, Trans
   private void updateEncumbranceToApplyTransaction(Transaction transaction, Transaction encumbrance, CurrencyUnit currency,
       Map<String, Transaction> existingTransactionMap) {
     MonetaryAmount expended = Money.of(encumbrance.getEncumbrance().getAmountExpended(), currency);
+    MonetaryAmount credited = Money.of(encumbrance.getEncumbrance().getAmountCredited(), currency);
     MonetaryAmount awaitingPayment = Money.of(encumbrance.getEncumbrance().getAmountAwaitingPayment(), currency);
     MonetaryAmount amount = Money.of(transaction.getAmount(), currency);
+    expended = expended.subtract(credited);
```

Comment:
and here too; you will adjust credited with CREDIT transactions and update amountCredited

Reply:
Thanks, fixed